src/Gui/Document.cpp

```diff
@@ -1476,5 +1477,22 @@ void Document::handleChildren3D(ViewProvider* viewProvider)
                 }
             }
         }
+    } else if (viewProvider && viewProvider->isDerivedFrom(ViewProviderDocumentObjectGroup::getClassTypeId())) {
+
+        ViewProviderDocumentObject* vp = static_cast<ViewProviderDocumentObject *>(viewProvider);
+        auto* ext = vp->getObject()->getExtensionByType<App::GroupExtension>();

```

Comment:
It seems you don't need the ViewProvider extension, but the DocumentObject extension. Maybe you shpould change the initial check in the else if clause fot the DocumentObject extension. If not than the check must happen here. Currently ext could be nullptr (there is a minimal chance, but coder erros could happen).